pkg/api/types.go

```diff
@@ -916,29 +916,62 @@ type Service struct {
 	Status ServiceStatus `json:"status,omitempty"`
 }
 
-// Endpoints is a collection of endpoints that implement the actual service, for example:
-// Name: "mysql", Endpoints: [{"ip": "10.10.1.1", "port": 1909}, {"ip": "10.10.2.2", "port": 8834}]
+// Endpoints is a collection of endpoints that implement the actual service.  Example:
+//   Name: "mysvc",
+//   Subsets: [
+//     {
+//       Addresses: [{"ip": "10.10.1.1"}, {"ip": "10.10.2.2"}],
+//       Ports: [{"name": "a", "port": 8675}, {"name": "b", "port": 309}]
+//     },
+//     {
+//       Addresses: [{"ip": "10.10.3.3"}],
+//       Ports: [{"name": "a", "port": 93}, {"name": "b", "port": 76}]
+//     },
+//  ]
```

Comment:
And I assume there is no intersection between the Cartesian products of each `EndpointSubset` value otherwise it is an error? Specifically, an IP address must not occur in more than one `EndpointSubset`.
Is it fine to have instead of this:

```
Subsets: [
  {
     Addresses: [{"ip": "10.10.1.1"}, {"ip": "10.10.2.2"}],
     Ports: [{"name": "a", "port": 8675}, {"name": "b", "port": 309}]
    },
]
```

to have this:

```
Subsets: [
  {
     Addresses: [{"ip": "10.10.1.1"}],
     Ports: [{"name": "a", "port": 8675}, {"name": "b", "port": 309}]
    },
    {
     Addresses: [{"ip": "10.10.2.2"}],
     Ports: [{"name": "a", "port": 8675}, {"name": "b", "port": 309}]
    },
]
```

since some algebra tells us that these should be equivalent?